### PR TITLE
Add custom domain support to callable functions

### PR DESCRIPTION
**Pull request by samtstern**
Hey there! So you want to contribute to a Firebase SDK?
Before you file this pull request, please read these guidelines:

### Discussion

[Googlers: see API review here](https://docs.google.com/document/d/1cSIcGCW2QR1OLHX4RDCVNhpBUolnG6tfNpJBR2pn1aM/edit#)

JS implementation for reference:
https://github.com/firebase/firebase-js-sdk/pull/3825

### Testing

  * Added new unit tests

### API Changes

  * See above

### CODE REVIEW

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (samtstern) [inline Functions/FirebaseFunctions/FIRFunctions.m:129]: Just want to make sure, this doesn't break the public API right?

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (paulb777) [inline Functions/FirebaseFunctions/FIRFunctions.m:142]: This is either a typo or needs to be made clearer.

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h
Comment (paulb777) [inline Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h:59]: There is no NS_SWIFT_NAME. Remove the clang-format off.

File: Functions/README.md
Comment (paulb777) [inline Functions/README.md:15]: Thanks

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (paulb777) [reply Functions/FirebaseFunctions/FIRFunctions.m:129]: As long as there's not a corresponding change in a public .h file

File: Functions/Example/Tests/FIRFunctionsTests.m
Comment (morganchen12) [inline Functions/Example/Tests/FIRFunctionsTests.m:48]: These can be pulled into the test setup method to avoid repeating them in each test. You'll need to add a variable to the test class to store the `functions` object.

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (samtstern) [reply Functions/FirebaseFunctions/FIRFunctions.m:142]: Typo!

File: Functions/Example/Tests/FIRFunctionsTests.m
Comment (samtstern) [reply Functions/Example/Tests/FIRFunctionsTests.m:48]: Good idea, thanks.

File: Functions/CHANGELOG.md
Comment (morganchen12) [inline Functions/CHANGELOG.md:2]: Release notes should be past tense, and use `class method` instead of `constructor`.

Comment (samtstern) [reply Functions/CHANGELOG.md:2]: But it's a constructor, isn't it?  I was following the wording from the release note about setting region.

Comment (morganchen12) [reply Functions/CHANGELOG.md:2]: Alright then, keep your constructors. But do fix the spelling!

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (morganchen12) [inline Functions/FirebaseFunctions/FIRFunctions.m:138]: Undo this change?

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (morganchen12) [inline Functions/FirebaseFunctions/FIRFunctions.m:142]: missing `]`

**Review by morganchen12 — APPROVED** [review]
LGTM with some nits, but please wait for Paul's approval and green CI before merging.

File: Functions/CHANGELOG.md
Comment (paulb777) [inline Functions/CHANGELOG.md:1]: Next version will be 7.1.0

File: Functions/FirebaseFunctions/FIRFunctions.m
Comment (paulb777) [reply Functions/FirebaseFunctions/FIRFunctions.m:138]: +1

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h
Comment (paulb777) [inline Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h:51]: Would be nice to verify if this is still really needed, but perhaps ok to deal with separately along with the other existing disables? @ryanwilson any thoughts here.

File: Functions/CHANGELOG.md
Comment (samtstern) [reply Functions/CHANGELOG.md:1]: Time flies!

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h
Comment (samtstern) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h:51]: I think it is.  I took it out and the formatter moved by `SWIFT_NAME` thing

**Comment by paulb777** [comment]
+ @ryanwilson for API review

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h
Comment (ryanwilson) [inline Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h:53]: Can developers provide both a region and custom domain, or are they mutually exclusive?

Comment (samtstern) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h:53]: They are mutually exclusive as `region` has no meaning when `customDomain` is set, the URL simply becomes `${customDomain}/${functionName}`

File: Functions/CHANGELOG.md
Comment (samtstern) [reply Functions/CHANGELOG.md:1]: Good catch!

File: Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h
Comment (ryanwilson) [inline Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h]: Nit: you can remove it here as well.

Comment (ryanwilson) [reply Functions/FirebaseFunctions/Public/FirebaseFunctions/FIRFunctions.h]: sorry was looking at an old revision, deleting

**Review by ryanwilson — APPROVED** [review]

**Pull request merged**
